src/lib.rs

```diff
@@ -1,35 +1,109 @@
+    }
 
-    // Info/Query connection
-    fn info(&mut self, req: &RequestInfo) -> ResponseInfo;
+    // Query Connection: Query your application. This usually resolves through a merkle tree holding
+    // the state of the app.
+    fn query(&mut self, _req: &RequestQuery) -> ResponseQuery {
+        ResponseQuery::new()
+    }
 
-    fn set_option(&mut self, req: &RequestSetOption) -> ResponseSetOption;
+    // Mempool Connection:  Used to validate incoming transactions.  If the application reponds
+    // with a non-zero value, the transaction is added to Tendermint's mempool for processing
+    // on the deliver_tx call below.
+    fn check_tx(&mut self, _req: &RequestCheckTx) -> ResponseCheckTx {
+        ResponseCheckTx::new()
+    }
 
-    fn query(&mut self, p: &RequestQuery) -> ResponseQuery;
+    // Consensus Connection:  Called once on startup. Usually used to establish initial (genesis)
+    // state.
+    fn init_chain(&mut self, _req: &RequestInitChain) -> ResponseInitChain {
+        ResponseInitChain::new()
+    }
 
-    // Mempool connection
-    fn check_tx(&mut self, p: &RequestCheckTx) -> ResponseCheckTx;
+    // Consensus Connection: Called at the start of processing a block of transactions
+    // The flow is:
+    // begin_block()
+    //   deliver_tx()  for each transaction in the block
+    // end_block()
+    // commit()
+    fn begin_block(&mut self, _req: &RequestBeginBlock) -> ResponseBeginBlock {
+        ResponseBeginBlock::new()
+    }
 
-    // Consensus connection
-    fn init_chain(&mut self, p: &RequestInitChain) -> ResponseInitChain;
+    // Consensus Connection: Actually processing the transaction, performing some form of a
+    // state transistion.
+    fn deliver_tx(&mut self, _p: &RequestDeliverTx) -> ResponseDeliverTx {
+        ResponseDeliverTx::new()
+    }
 
-    fn begin_block(&mut self, p: &RequestBeginBlock) -> ResponseBeginBlock;
+    // Consensus Connection: Called at the end of the block.  Often used to update the validator set.
+    fn end_block(&mut self, _req: &RequestEndBlock) -> ResponseEndBlock {
+        ResponseEndBlock::new()
+    }
 
-    fn deliver_tx(&mut self, p: &RequestDeliverTx) -> ResponseDeliverTx;
+    // Consensus Connection: Commit the block with the latest state from the application.
+    fn commit(&mut self, _req: &RequestCommit) -> ResponseCommit {
+        ResponseCommit::new()
+    }
+}
 
-    fn end_block(&mut self, p: &RequestEndBlock) -> ResponseEndBlock;
+// Setup the app and start the server using localhost and default tendermint port 26658
+pub fn run_local<A>(app: A)
+where
+    A: Application + 'static + Send + Sync,
+{
+    let addr = "127.0.0.1:26658".parse().unwrap();
+    run(addr, app);
+}
 
-    fn commit(&mut self, p: &RequestCommit) -> ResponseCommit;
-}
\ No newline at end of file
+// Setup the application and start the server. Use this fn when setting different ip:port.
+pub fn run<A>(listen_addr: SocketAddr, app: A)
```

Comment:
Maybe put `app` first, because that's its positioning in `run_local`: the most significant argument should usually be first.